lib/primitiv.gi

```diff
@@ -40,14 +43,35 @@ local s,fname,ind;
 end);
 
 BindGlobal("PRIMGrp",function(deg,nr)
-  PrimGrpLoad(deg);
+  local filename,strm,r,l;
   if nr>PRIMLENGTHS[deg] then
     Error("There are only ",PRIMLENGTHS[deg]," groups of degree ",deg,"\n");
   fi;
+  if deg > 4095 then
+    filename := Concatenation(GAPInfo.PackagesInfo.primgrp[1].InstallationPath,"/data/ExtendedPrimitiveGroupsData/PrimitiveGroups_", String(deg),"_", String(nr), ".g.gz");
+    strm:=InputTextFile(filename);;
+    if strm = fail then
+      Error("Primitive group of degree ", deg, " with id ", nr, " not found! Note that primitive groups of degree 4096 to 8191 must be downloaded separately. They can be obtained from https://doi.org/10.5281/zenodo.10411367\n");
+    fi;
+    r:=EvalString(ReadAll(strm));;
+    CloseStream(strm);;
+    if not "name" in RecNames(r) then
+      r.name:="";
+    fi;
+    if IsString(r.size) then
+      r.size:=EvalString(r.size);
```

Comment:
Generally we have been trying to minimize the places using `EvalString`. Why is this necessary here?

Reply:
Looking at the data files, this seems to be (only?) for handling the symmetric and alternating groups with strings like `"Factorial(6228)/2"`. Are there any other cases? If it is just those, I would suggest to just detect these cases. (Actually, I would not have stored these cases at all, as it is trivial to generate them at runtime :-)

Reply:
I removed the "" around "Factorial(6228)/2" so that it evaluates immediately (similar to the original data for the primitive groups of degree less than 4096)